client/src/app/pages/applications/applications-table-assessment/applications-table-assessment.tsx

```diff
@@ -473,80 +268,210 @@ export const ApplicationsTable: React.FC = () => {
+        }) + "...",
+      type: FilterType.multiselect,
+      selectOptions: [
+        { key: "source", value: "Source" },
+        { key: "maven", value: "Maven" },
+        { key: "proxy", value: "Proxy" },
+      ],
+      getItemValue: (item) => {
+        const searchStringArr: string[] = [];
+        item.identities?.forEach((appIdentity) => {
+          const matchingIdentity = identities.find(
+            (identity) => identity.id === appIdentity.id
           );
-        }
-      },
-      (error: AxiosError) => {
-        pushNotification({
-          title: getAxiosErrorMessage(error),
-          variant: "danger",
+          searchStringArr.push(matchingIdentity?.kind || "");
         });
-      }
-    );
-  };
+        const searchString = searchStringArr.join("");
+        return searchString;
+      },
+    },
+    {
+      key: "repository",
+      title: t("terms.repositoryType"),
+      placeholderText:
+        t("actions.filterBy", {
+          what: t("terms.repositoryType").toLowerCase(),
+        }) + "...",
+      type: FilterType.select,
+      selectOptions: [
+        { key: "git", value: "Git" },
+        { key: "subversion", value: "Subversion" },
+      ],
+      getItemValue: (item) => item?.repository?.kind || "",
+    },
+    {
+      key: "binary",
+      title: t("terms.artifact"),
+      placeholderText:
+        t("actions.filterBy", {
+          what: t("terms.artifact").toLowerCase(),
+        }) + "...",
+      type: FilterType.select,
+      selectOptions: [
+        { key: "binary", value: t("terms.artifactAssociated") },
+        { key: "none", value: t("terms.artifactNotAssociated") },
+      ],
+      getItemValue: (item) => {
+        const hasBinary =
+          item.binary !== "::" && item.binary?.match(/.+:.+:.+/)
+            ? "binary"
+            : "none";
 
-  const { data: reviewAssessmentSetting } = useSetting(
-    "review.assessment.required"
-  );
+        return hasBinary;
+      },
+    },
+    {
+      key: "tags",
+      title: t("terms.tags"),
+      type: FilterType.multiselect,
+      placeholderText:
+        t("actions.filterBy", {
+          what: t("terms.tagName").toLowerCase(),
+        }) + "...",
+      getItemValue: (item) => {
+        let tagNames = item?.tags?.map((tag) => tag.name).join("");
+        return tagNames || "";
+      },
+      selectOptions: dedupeFunction(
+        tagCategories
+          ?.map((tagCategory) => tagCategory?.tags)
+          .flat()
+          .filter((tag) => tag && tag.name)
+          .map((tag) => ({ key: tag?.name, value: tag?.name }))
+      ),
+    },
+  ];
 
-  const reviewSelectedRows = () => {
-    if (selectedRows.length !== 1) {
-      const msg = "The number of applications to be reviewed must be 1";
-      pushNotification({
-        title: msg,
-        variant: "danger",
-      });
-      return;
-    }
+  const tableControls = useLocalTableControls({
+    idProperty: "name",
```

Comment:
This is where I meant to post the other comment:

> One more thought: Should we use `idProperty: "id"` instead of `"name"` here? Do we need it to be a string (idProperty can refer to either a string field or a number field)? Are app names guaranteed to be unique?